src/native/http_request_response.c

```diff
@@ -289,36 +387,28 @@ static void s_on_incoming_body_fn(
     // Other threads might edit the callback struct, so ensure that we gain a lock on it
     aws_mutex_lock(&callback->lock);
 
-    JNIEnv *env = aws_jni_get_thread_env(callback->connection->jvm);
+    struct aws_byte_cursor body_in_remaining = *data;
+    size_t total_window_increment = 0;
 
-    jobject jByteBuffer = aws_jni_byte_buffer_copy_from_cursor(env, data);
-    jint window_increment = (*env)->CallIntMethod(
-        env,
-        callback->java_crt_http_callback_handler,
-        s_crt_http_stream_handler.onResponseBody,
-        callback->java_http_stream,
-        jByteBuffer);
+    while (body_in_remaining.len > 0) {
```

Comment:
If you are transferring less than the buffer size, how will that ever get published to Java?

Reply:
I force a java.onRespBody() publish when native.onStreamComplete() is called even if buffer is not full.